### ISSUE TITLE
Style/TrailingCommaInHashLiteral and Layout/MultilineHashBraceLayout safe autocorrects produce invalid file

### ISSUE THREAD

**Issue by NeoPhi** [rubocop/rubocop#11828]
When running safe autocorrect on a file with `Style/TrailingCommaInHashLiteral` and `Layout/MultilineHashBraceLayout` violations the resulting file is no longer valid Ruby due to incorrect handling of an end of line comment.

## Expected behavior

The file is correctly auto formatted.

## Actual behavior

```
% rubocop --debug -a sample.rb
For /Users/danielr/Temp/rubocop: Default configuration from /Users/danielr/.asdf/installs/ruby/2.7.8/lib/ruby/gems/2.7.0/gems/rubocop-1.50.2/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/danielr/Temp/rubocop/sample.rb
Loading cache from /Users/danielr/.cache/rubocop_cache/c341ad2e58488ba1da1294a308c31ada931177bd/6d7a3b621ca1730e04accd938619e4bdab66cfb1/d6271005d8edd8d4284ca300de9e11a434b19bb1
F

Offenses:

sample.rb:5:27: C: [Corrected] Style/TrailingCommaInHashLiteral: Avoid comma after the last item of a hash.
               'b' => true, # EOL B
                          ^
sample.rb:6:3: C: [Corrected] Layout/MultilineHashBraceLayout: Closing hash brace must be on the same line as the last hash element when opening brace is on the same line as the first hash element.
  },
  ^
sample.rb:6:3: F: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
  blocklist: { 'c' => true }
  ^^^^^^^^^

1 file inspected, 3 offenses detected, 2 offenses corrected
Finished in 0.17126799980178475 seconds```

## Steps to reproduce the problem

Input `sample.rb` file:

```
# frozen_string_literal: true

CONFIG = {
  allowlist: { 'a' => true, # EOL A
               'b' => true, # EOL B
  },
  blocklist: { 'c' => true }
}.freeze
```

## RuboCop version

```
% rubocop -V
1.50.2 (using Parser 3.2.2.1, rubocop-ast 1.28.0, running on ruby 2.7.8) [x86_64-darwin22]
```

**Comment by alexeyschepin** [comment]
The bug is in `Layout/MultilineHashBraceLayout` cop. 

## Steps to narrow down the problem
Run `rubocop --only Layout/MultilineHashBraceLayout -a  sample.rb` on

`sample.rb`
```ruby
{
  allowlist: { 'a' => true, # EOL A
               'b' => true, # EOL B
  },
  blocklist: { 'c' => true }
}
```

It will be autocorrected into

```ruby
{
  allowlist: { 'a' => true, # EOL A
               'b' => true,} # EOL B,
  blocklist: { 'c' => true }
}
```

The issue must be related to nested properties.
